src/syscall_table/aarch64.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+// This file is auto-generated by `tools/generate_syscall_tables`.
+// Do NOT manually edit!
+// Generated at: Fri Jul 30 10:20:08 EEST 2021
+// Kernel version: 5.10
+
+use std::collections::HashMap;
+
+pub(crate) fn make_syscall_table(map: &mut HashMap<String, i64>) {
```

Comment:
There is no need to pass a mutable HashMap. We are creating a static HashMap, and we can just return it at the end of the function instead of calling subsequent `insert` functions.

Also, since we are creating a static HashMap, instead of using insert, we should use the recommend way as defined here: https://doc.rust-lang.org/std/collections/struct.HashMap.html#examples in the example with the fixed list of elements.

I also believe it is easier to work with `static str` instead of String. Throughout the code we are initializing the HashMap using a static string, then when we are retrieving the keys we are also using a static string. There is no need to use dynamic allocated strings in this scenario. It makes the code more verbose than it needs to be.

Reply:
When building this on x86, it returns a warning:
```
 --> src/syscall_table/aarch64.rs:9:19
  |
9 |     pub(crate) fn make_syscall_table() -> HashMap<&'static str, i64> {
  |                   ^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default
```

We should add a test to check for warnings in the custom pipeline, and fix this warning.

Reply:
It no longer reproduces for me either. It might have been a local change that I had.